From 49054c3ed14841d8a8c7ff8a4eeef40c01441664 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sun, 29 Dec 2019 16:06:41 +0200
Subject: [PATCH] [SW-6739] arcboot: increase stack and heap

increase stack and heap allocation because the of
synopsis secureboot library, hash and decryption
routines require more memory then previous allocation

Change-Id: Iddb3df8e2600f8384b226e1a4c987e24aa026855
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/goya2_arc/arcboot_rom.lds b/goya2_arc/arcboot_rom.lds
index ddc927a..4c9b062 100644
--- a/goya2_arc/arcboot_rom.lds
+++ b/goya2_arc/arcboot_rom.lds
@@ -86,7 +86,7 @@
 	.heap : {
 		. = ALIGN(32 / 8);
 		__start_heap = . ;
-		. = . + (DEFINED(__HEAP_SIZE) ? __HEAP_SIZE : (DEFINED(__DEFAULT_HEAP_SIZE) ? __DEFAULT_HEAP_SIZE : 1k)) ;
+		. = . + (DEFINED(__HEAP_SIZE) ? __HEAP_SIZE : (DEFINED(__DEFAULT_HEAP_SIZE) ? __DEFAULT_HEAP_SIZE : 4k)) ;
 		__end_heap = . ;
 		. = ALIGN(0x8);
 	} > REGION_HEAP
@@ -94,7 +94,7 @@
 	.stack : {
 		. = ALIGN(32 / 8);
 		__stack = . ;
-		. = . + (DEFINED(__STACK_SIZE) ? __STACK_SIZE : (DEFINED(__DEFAULT_STACK_SIZE) ? __DEFAULT_STACK_SIZE : 512)) ;
+		. = . + (DEFINED(__STACK_SIZE) ? __STACK_SIZE : (DEFINED(__DEFAULT_STACK_SIZE) ? __DEFAULT_STACK_SIZE : 9k)) ;
 		__stack_top = . ;
 	} > REGION_STACK
 }
\ No newline at end of file
